eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -297,15 +297,23 @@ else if (inside.size() > 32 && !(inside instanceof SetIterable))
      *
      * @since 1.0
      */
-    boolean containsAllIterable(Iterable<?> source);
+    default boolean containsAllIterable(Iterable<?> source)
+    {
+        RichIterable<?> outside = this.toSet();
+        return StreamSupport.stream(source.spliterator(), false).allMatch(outside::contains);
+    }
```

Comment:
If this is a set, then calling toSet() is unnecessary, and this implementation is slower than necessary. I think we should leave this abstract on RichIterable.